src/gameplayv2/resistance/components/AssassinateComponent.ts

```diff
@@ -0,0 +1,102 @@
+import Game from "../../Game";
+
+export class AssassinateComponent {
+    protected assassin: string;
+    protected isEnabled: boolean;
+    protected role: string = 'Assassin';
+    protected alliance: string = 'Spy';
+    protected specialPhase: string = 'assassination';
+    protected description: string =  'If the resistance win 3 missions, the Assassin can shoot one person for Merlin, or two people for Tristan and Isolde. If they are correct, the spies win!';
+    protected orderPriorityInOptions: number = 90;
+    protected playerShot: string = '';
+    protected playerShot2: string = '';
+    protected thisRoom: Game;
+    constructor(assassin: string, isEnabled: boolean, thisroom: Game) {
+        this.assassin = assassin;
+        this.isEnabled = isEnabled;
+        this.thisRoom = thisroom;
+    }
+
+    getMorgana(): string {
+        return this.assassin;
+    }
+
+    setViewer(newAssassin: string): void {
+        this.assassin = newAssassin;
+    }
+
+    getEnabled(): boolean {
+        return this.isEnabled;
+    }
+
+    setEnabled(newEnabled: boolean): void {
+        this.isEnabled = newEnabled;
+    }
+
+    //todo: Change this to be much more descriptive
+    seeSpies(): string[]{
+        // Assassin sees all spies except oberon
+        if (this.thisRoom.gameStarted === true) {
+            let spies:string[] = [];
+            for (let i = 0; i < this.thisRoom.playersInGame.length; i++) {
+                if (this.thisRoom.playersInGame[i].alliance === 'Spy') {
+                    if (this.thisRoom.playersInGame[i].role === 'Oberon') {
+                    // don't add oberon
+                        } else {
+                            // add the spy
+                            spies.push(this.thisRoom.playersInGame[i].username);
+                        }
+                    }
+                }
+                return spies;
+            }
+        }
+
+    //TO DO: can this possibly become a phase component?
+    checkSpecialMove(): boolean{
```

Comment:
Yep this is one of the parts in the existing game engine which I wanted to redesign. This part essentially checks when we should enter assassination phase. 

See this line:
```
this.thisRoom.phase = this.specialPhase;
```

We should decide how to make this state machine work and better define how transitions in the game engine should be handled.